src/store/actions/thunk.ts

```diff
@@ -0,0 +1,48 @@
+import { Dispatch } from '@reduxjs/toolkit';
+import axios, { AxiosError } from 'axios';
+import { Videos } from '../../components/VideoList/types';
+import { BASE_FETCH_URL, API_KEY, MAX_RESULTS } from '../../constants/api';
+import { errorActions } from '../slices/error';
+import { loadingActions } from '../slices/loading';
+import { videosActions } from '../slices/videos';
+
+export function getVideos(
+  query: string,
+  nextPageToken: void
+): (dispatch: Dispatch) => Promise<void>;
+
+export function getVideos(
+  query: string,
+  nextPageToken: string
+): (dispatch: Dispatch) => Promise<void>;
+
+export function getVideos<P>(query: string, nextPageToken: P) {
+  return async (dispatch: Dispatch) => {
+    loadingActions.setIsLoading(true);
```

Comment:
Can you use `createAsyncThunk`, it will handle all the loading state for you.

Reply:
Sure